plugins/module_utils/foreman_helper.py

```diff
@@ -1445,7 +1453,10 @@ def _flatten_entity(entity, foreman_spec):
             flat_name = spec.get('flat_name', key)
             property_type = spec.get('type', 'str')
             if property_type == 'entity':
-                result[flat_name] = value['id']
+                if value:
```

Comment:
should this be kept, or become `if not isinstance(value, NoEntity)`? (then it also doesn't need the `__bool__`).

Reply:
`    "msg": "Failed to ensure entity state: TypeError: 'NoEntity' object has no attribute '__getitem__'"`

This answers my question, I guess.

Reply:
(that's Python2, if you wonder)

Reply:
I thought about not even instantiating the NoEntity, but using the class (as a singleton). And would `if value is NoEntity:` work?